### ISSUE TITLE
Reduce the required dependencies in PatientLevelPrediction

### ISSUE THREAD

**Issue by ablack3** [OHDSI/PatientLevelPrediction#199]
**Problem**
The PatientLevelPredication package provides large set of functionality but comes with a lot of dependencies which can make the package difficult to install. In particular xgboost requires system dependencies that are not always available. Some users would like to simply view a PLP results shiny app and do not need all of the functionality that comes with PLP, for example when the model was fit using the Arachne Execution Engine in Atlas. However (as far as I can tell) the PLP results Shiny apps require the PatientLevelPrediction package to be installed. More generally, if a user has no specific use case for tensorflow, xgboost, or python (reticulate) it would be helpful if they had the option to only install the components of the PLP package that they needed.

**Desired Solution**
Move xgboost, tensorflow, keras, and reticulate from Imports to Suggests in the DESCRIPTION file. When functions requiring these dependencies are called the function will check to see if the dependency is installed and if not instruct the user to install it.

```
my_fun <- function(a, b) {
  if (!requireNamespace("pkg", quietly = TRUE)) {
    stop("Package \"pkg\" needed for this function to work. Please install it.",
      call. = FALSE)
  }
}
```
From: https://r-pkgs.org/description.html?q=depend#dependencies

**Alternative Solution**
As an alternative I could look into why PLP results shiny apps produced by the Atlas execution engine require the PatientLevelPrediction Namespace. It seems like these could be stand-alone shiny apps that do not rely on the PLP package to run.

**Comment by jreps** [comment]
xgboost has been moved in the 'development' branch.  The shiny app can run without PLP - see: https://github.com/OHDSI/ShinyDeploy/tree/master/PCEvalidation - the shinyDeploy doesn't have PLP installed - so this version of code doesnt require any install (any PLP result in shinyDeploy has code without PLP as a dependancies).

I'm planning to make other dependancies conditional in the future- although apart from xgboost nothing else seems to have issues at the moment.  Also - the issue with xgboost (seems to be a linux specific issue) is that people need to make sure they can compile things, this issue is fixable and probably worth fixing as gradient boosting is a good method.

**Comment by jreps** [comment]
I've been checking the package and I think I can:

Remove these imports completely: doParallel, bit, ff, foreach, RCurl, RJSONIO, viridisLite, Methods

Move these imports suggests: Tensorflow (only used by CIReNN.R), survAUC (only used by survival models), Officer/Diagram  (only used by report generation), Keras (used by CIReNN.R, CovNN.R, CovNN2.R, DeepNN.R plus predict and save/load), htmltools (need to check this is used in the shiny app), plotly (only used in the shiny app)

This brings imports to < 30. I'm testing that this doesn't break the tests and will then commit the update.

I may also be able to remove: AUC (this is used as it was faster than the internal PLP calculator but I think that was due to a bug - so I will check speed and if it is no longer faster, I'll remove it), cowplot (this is only used in smooth calibration plot and I have an alternative I can use) and Hmisc (this is used when running metrics for survival models but I can easily write code to replace the use of this package).

Reticulate is used a lot - so I'd prefer to keep that in the imports.  PLP functionality is pretty limited without reticulate.

**Comment by ablack3** [comment]
Wow thanks for looking into this @jreps! I should figure out why I was getting error with running true PLP shiny viewer app without PLP installed. Let me know if I can help.
